etc/bash_completion.d/dnf

```diff
@@ -34,126 +132,251 @@ _is_path()
+        ;;
+
+    info|list)
+        case "$prev" in
+            --installed|installed)
+            _dnf_show_packages list installed ;;
+
+            --available|available)
+            _dnf_show_packages list available ;;
+
+            --updates|--upgrades|updates|upgrades)
+            _dnf_show_packages list upgrades ;;
+
+            *)
+            complete_commands="all available installed extras updates
+                               upgrades autoremove recent"
+        esac
+
+        extra_options="--all --available --installed --extras --updates
+                           --upgrades --autoremove --recent"
+        ;;
+
+    info-*|list-*|updateinfo|summary-updateinfo)
+        extra_options="--summary --list --info"
+        complete_commands="summary list info"
+        ;;
+
+    in|install*|localinstall)
+        _dnf_show_packages $command
+        ;;
+
+    mark)
+        _dnf_show_packages list installed
+        ;;
+
+    mc|ref|makecache|refresh)
+        [[ "$prev" == *"timer" ]] && return
+        extra_options="--timer"
+        complete_commands="timer"
+        ;;
+
+    provides)
+        ;;
+
+    ri|rei|reinstall)
+        _dnf_show_packages $command
+        ;;
+
+    rm|remove*|erase*)
+        _dnf_show_packages $command
+        extra_options="--oldinstallonly --duplicates --duplicated"
+        ;;
+
+    repoinfo|repolist)
+        extra_options="--all --enabled --disabled"
+        case $prev in
+        enabled|--enabled)
+            complete_commands="$( _dnf_commands_helper $command enabled "$cur" )"
             ;;
-        *)
+        disabled|--disabled)
+            complete_commands="$( _dnf_commands_helper $command disabled "$cur" )"
             ;;
+        all|--all|*)
+            complete_commands="$( _dnf_commands_helper $command all "$cur" )"
+        esac
+        ;;
+
+    repoquery*)
+        extra_options="$__dnf_repoquery_subcmds"
+        ;;
+
+    repo*-pkgs|repo*-packages)
+        complete_commands="$__dnf_repopkgs_subcmds"
+        ;;
+
+    se|search)
+        [[ "$prev" == *"all" ]] && return
+        extra_options="--all"
+        complete_commands="all"
+        ;;
+
+    sh|shell) ;;
+
+    swap) ;;
+
+    um|u-m|upgrade|upgrade-n*)
+        _dnf_show_packages $command
+        ;;
+
+    up|up-min|update|update-n*)
+        _dnf_show_packages $command
+        ;;
+
+    whatprovides) ;;
+
+    *)
+        complete_commands="$cmd_list"
+        ;;
     esac
 
     $split && return
 
-    local comp
-    local cache_file="/var/cache/dnf/packages.db"
-    local sqlite3="sqlite3 -batch -init /dev/null"
-    if [[ $command ]]; then
-
-        case $command in
-            install|update|upgrade|reinstall|info)
-                if ! _is_path "$cur"; then
-                    if [ -r $cache_file ] && ! _modified_sack words[@]; then
-                        COMPREPLY=( $( compgen -W '$( $sqlite3 $cache_file "select pkg from available WHERE pkg LIKE \"$cur%\"" 2>/dev/null )' ) )
```

Comment:
not using of this cache database really slows down completion of the package name (the same for erase/remove/downgrade commands)